packages/labs/src/components/select/suggest.tsx

```diff
@@ -0,0 +1,265 @@
+                        {...htmlInputProps}
+                        inputRef={this.refHandlers.input}
+                        onChange={this.handleQueryChange}
+                    />
+                </div>
+                <div onKeyDown={handleKeyDown} onKeyUp={handleKeyUp}>
+                    <Menu ulRef={listProps.itemsParentRef}>
+                        {this.renderItems(listProps)}
+                    </Menu>
+                </div>
+            </Popover>
+        );
+    }
+
+    private renderItems({ activeItem, filteredItems, handleItemSelect }: IQueryListRendererProps<T>) {
+        const { itemRenderer, noResults } = this.props;
+        if (filteredItems.length === 0) {
+            return noResults;
+        }
+        return filteredItems.map((item, index) => itemRenderer({
+            index,
+            item,
+            handleClick: (e) => handleItemSelect(item, e),
+            isActive: item === activeItem,
+        }));
+    }
+
+    private selectText = () => {
+        if (this.input != null) {
+            this.input.focus();
+            // wait until the input is focused to select the text inside of it
+            requestAnimationFrame(() => this.input.setSelectionRange(0, this.input.value.length));
+        }
+    }
+
+    private handleInputClick = (_event: React.SyntheticEvent<HTMLElement>) => {
+        this.selectText();
+    }
+
+    private handleActiveItemChange = (activeItem: T) => this.setState({ activeItem });
+
+    private handleItemSelect = (item: T, event: React.SyntheticEvent<HTMLElement>) => {
+        this.selectText();
+
+        this.setState({
+            isTyping: false,
+            query: "",
+            selectedItem: item,
+        });
+
+        Utils.safeInvoke(this.props.onItemSelect, item, event);
+    }
+
+    private handlePopoverInteraction = (nextOpenState: boolean) => requestAnimationFrame(() => {
+        const { closeOnSelect, popoverProps = {} } = this.props;
+
+        if (this.input == null) {
+            return;
+        }
+
+        if (this.input !== document.activeElement) {
+            // the input is no longer focused so we can close the popover
+            this.input.blur();
+            this.setState({ isOpen: false });
+        } else {
+            this.setState({ isOpen: closeOnSelect ? nextOpenState : true });
+        }
+
+        Utils.safeInvoke(popoverProps.onInteraction, nextOpenState);
+    })
+
+    private handlePopoverDidOpen = () => {
+        const { popoverProps = {} } = this.props;
+
+        // scroll active item into view after popover transition completes and all dimensions are stable.
+        if (this.queryList != null) {
+            this.queryList.scrollActiveItemIntoView();
+        }
+
+        Utils.safeInvoke(popoverProps.popoverDidOpen);
+    }
+
+    private handlePopoverWillClose = () => {
+        const { popoverProps = {} } = this.props;
+
+        // resetting the query when the popover close
```

Comment:
We always reset on close here. If we don't, you can end up with the filtered list desynced from what's shown in the input (because we don't show `query` in the input at all times)